src/LineageTree/lineageTree.py

```diff
@@ -2485,28 +2485,32 @@ def __dp(dist_mat: np.ndarray, start_d: int=0, back_d: int=0, w: int=0, slope=No
         ):
             i = N - 1
             j = M - back_d + min_index2 - 1
+            final_cost = cost_mat[N, M - back_d + min_index2]
```

Comment:
Why do you have `i` and `j` declared and not used?

Reply:
i and j are used below to calculate the path
the final cost is the point where the path ends, if there's no delay, it's (N-1, M-1), but it can change when we add delay, so I created the variable inside the if